core/src/main/java/org/elasticsearch/indices/recovery/RecoveryTargetService.java

```diff
@@ -263,12 +263,6 @@ public RecoveryResponse newInstance() {
                 return;
             }
 
-            if (cause instanceof IndexShardClosedException) {
```

Comment:
I'm fine with removing it - it's useless . I'm ok with just checking it further after the PR is done.